### ISSUE TITLE
why can't predict occluding contours?

### ISSUE THREAD

**Issue by deng29** [MichaelRamamonjisoa/SharpNet#17]
Hello,
    I load the model pre-trained on PBRS, then finetune on NYU. But when I run demo.py, the model can't predict occluding contours, the predicted result is blank. However, the other two task can work normally. I'm confused......

**Comment by MichaelRamamonjisoa** [comment]
Hi,

Thanks for your interest in SharpNet. Could you share your test image? Also, if possible, can you share the trained weights? When finetuning on NYU, because they are not supervised anymore, the quality of contour decreases a bit.
However I am surprised that you cannot detect any contour anymore.

**Comment by deng29** [comment]
Hello,

I just use the official test split of NYU to test, the model training is completely in accordance with your instructions on github.

**Comment by deng29** [comment]
Hello,
To run eval.py for test, it need Eigen/Laina/Fu/Jiao prediction.mat, can you provide with me?
Sincerely thanks.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "MichaelRamamonjisoa/SharpNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年5月7日(星期五) 晚上9:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [MichaelRamamonjisoa/SharpNet] why can't predict occluding contours? (#17)





 
Hi,
 
Thanks for your interest in SharpNet. Could you share your test image? Also, if possible, can you share the trained weights? When finetuning on NYU, because they are not supervised anymore, the quality of contour decreases a bit.
 However I am surprised that you cannot detect any contour anymore.

**Comment by MichaelRamamonjisoa** [comment]
Hi, 

**Regarding occluding contour predictions**
It sounds like the finetuning was done for too many epochs. It may be a mistake on my part regarding the number of epochs used for finetuning (80 seems quite high). I'll try to investigate it some time later.
In the meantime if you want to check results I suggest you use the final weights I had obtained.
Finally, if your priority is good occlusion boundary predictions, I would suggest to use the PBRS weights. Indeed as I mentioned, since they are not supervised during fine-tuning, their quality will progressively decrease. They only serve as guidance for the early stages of finetuning depth branch to keep the sharpness obtained with synthetic data.

**Regarding other methods results**
For Eigen, check [here](https://cs.nyu.edu/~deigen/dnl/)
For Laina, check [here](https://github.com/iro-cp/FCRN-DepthPrediction#results)
For Fu (DORN), check [here](https://drive.google.com/file/d/1uRqOkCbJLwHWyx4oz19N6MQgrOSZQo6H/view)
For Jiao, I've put their predictions [here](https://drive.google.com/file/d/1Qpj5uffsFc4JliLQhMAovTSg7kh_tlkq/view?usp=sharing).

**Comment by deng29** [comment]
Hello,
Thank you for helping me solve my confusion.&nbsp;By the way, have you considered using semantic segmentation to assist depth estimation?&nbsp;I see some papers do this，but they rarely provide source code.&nbsp;I'm more interested in this direction, do you have done this kind of work?


Sincerely




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "MichaelRamamonjisoa/SharpNet"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年5月10日(星期一) 晚上9:56
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [MichaelRamamonjisoa/SharpNet] why can't predict occluding contours? (#17)





 
Hi,
 
Regarding occluding contour predictions
 It sounds like the finetuning was done for too many epochs. It may be a mistake on my part regarding the number of epochs used for finetuning (80 seems quite high). I'll try to investigate it some time later.
 In the meantime if you want to check results I suggest you use the final weights I had obtained.
 Finally, if your priority is good occlusion boundary predictions, I would suggest to use the PBRS weights. Indeed as I mentioned, since they are not supervised during fine-tuning, their quality will progressively decrease. They only serve as guidance for the early stages of finetuning depth branch to keep the sharpness obtained with synthetic data.
 
Regarding other methods results
 For Eigen, check here
 For Laina, check here
 For Fu (DORN), check here
 For Jiao, I've put their predictions here.

**Comment by MichaelRamamonjisoa** [comment]
Hi, let's discuss this offline, there are now several papers (with code, I hope?) that consider multi task supervision using semantic labels, but let's discuss this topic by email. Closing the issue for now

**Comment by deng29** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
